scripts/apps/archive/directives/MarkedItemTitle.js

```diff
@@ -0,0 +1,122 @@
+/**
+ * @module superdesk.apps.archive
+ * @ngdoc directive
+ * @name sdMarkedItemTitle
+ * @requires desks
+ * @requires authoring
+ * @requires highlightServce
+ * @requires $timeout
+ * @description
+ *   This directive is used in authoring-topbar to mark a story as a highlight or
+ *   to mark it for a desk.
+ */
+MarkedItemTitle.$inject = ['desks', '$timeout', 'authoring', 'highlightsService'];
+export function MarkedItemTitle(desks, $timeout, authoring, highlightsService) {
+    return {
+        scope: {
+            item: '=item',
+            markField: '@field'
+        },
+        templateUrl: 'scripts/apps/archive/views/marked_item_title.html',
+        // todo(petr): refactor to use popover-template once angular-bootstrap 0.13 is out
+        link: function(scope, elem) {
+            /*
+             * Toggle 'open' class on dropdown menu element
+             * @param {string} isOpen
+             */
+            scope.toggleClass = function(isOpen) {
+                scope.open = isOpen;
+            };
+
+            scope.marks = scope.item[scope.markField];
+            scope.hasMarkItemPrivilege = authoring.itemActions(scope.item).mark_item_for_desks;
+
+            scope.$on('item:' + scope.markField, ($event, data) => {
+                let marks = scope.marks || [];
+                let mark = data.mark_id;
+
+                if (scope.markField === 'marked_desks') {
+                    mark = {desk_id: data.mark_id};
+                }
+
+                if (scope.item._id === data.item_id) {
```

Comment:
If this condition is false, this even listener takes no action, so instead, you could again simplify and reduce indentation by placing this at the top of the function (on line 35):
```js
if (scope.item._id !== data.item_id) {
    return;
}
```
And then continue on without any more `if`'s